### DISCUSSION TITLE
Lag when calling i3status in shell script

### DISCUSSION THREAD

**Discussion by cpif** [i3/i3#5403] [Q&A] [answered]
Say I have a script like the following:

``` bash
#!/bin/bash
i3status -c stat.conf
```

And I specify that script as my status command in my i3 config file:

```
bar {
    status_command myscript
}
```

Everything works exactly the same as if I had this in my i3 config:

```
bar {
    status_command i3status
}
```

However, say I change my script to something like the following (all I'm doing is using sed to add an additional message to what i3status generates):

``` bash
#!/bin/bash
i3status -c stat.conf | sed 's/]$/,{"full_text":"MY MESSAGE"}]/'
```

For some reason, this causes i3bar to update at a much laggier pace. For example, if I raise or lower the volume, it does not update instantaneously. If I run the same script from a terminal, it spits out updates to `STDOUT` instantly when I adjust the volume.

I think there must be something stupidly obvious about what I'm doing wrong here. All I'd like is to be able to do a few small scripted manipulations of the output generated by i3status without losing its speediness.

Thanks!

**Comment by Airblader** [comment, marked as answer]
This is probably due to buffering. See eg https://stackoverflow.com/questions/11337041/force-line-buffering-of-stdout-in-a-pipeline

**Reply by cpif** [reply]
That was it -- thanks very much!
